tests/extras/test_logstash.py

```diff
@@ -1,7 +1,115 @@
-from yellowbox.extras.logstash import LogstashService
+from time import sleep
 
-# rebuilding the logstash service is an ongoing ticket, for now we just want to make sure it runs alright
+import pytest
+import json
+import socket
+from yellowbox.extras.logstash import FakeLogstashService
+
+@pytest.fixture
+def logstash():
+    ls = FakeLogstashService()
+    ls.start()
+    return ls
+
+
+def create_socket(logstash):
+    return socket.create_connection((logstash.local_host, logstash.port))
+
+
+def send_record(logstash, **kwargs):
+    s = create_socket(logstash)
+    s.sendall(json.dumps(kwargs).encode("utf-8")+b"\n")
+    s.close()
+    sleep(0.01)
+
+
+def send_records(logstash, *records):
+    s = create_socket(logstash)
+    data = "\n".join([json.dumps(record) for record in records]) + "\n"
+    data = data.encode("utf-8")
+    s.sendall(data)
+    s.close()
+    sleep(0.01)
+
+
+def test_sanity(logstash):
+    send_record(logstash, msg="test")
+    assert logstash.records[0] == {"msg": "test"}
+
+
+def test_multiple_records(logstash):
+    send_records(logstash, {"msg": "hello"}, {"msg": "meow"})
+    assert logstash.records == [{"msg": "hello"}, {"msg": "meow"}]
+
+
+def test_multiple_connections(logstash):
+    send_record(logstash, msg="test")
+    send_record(logstash, msg="test2")
```

Comment:
since these are different TCP connections, isn't there a risk if the messages appearing in reverse order?

Reply:
There is a very small risk, I believe the selector returns the events in order (that's how they're received in the OS).

Reply:
Actually scratch this, it's not possible, I've made sure of that. Send record sleeps.